packages/lesswrong/components/comments/NewUserGuidelinesDialog.tsx

```diff
@@ -0,0 +1,112 @@
+import Button from '@material-ui/core/Button';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import React from 'react';
+import { Link } from '../../lib/reactRouterWrapper';
+import { useNewEvents } from '../../lib/events/withNewEvents';
+import { Components, registerComponent } from '../../lib/vulcan-lib';
+import { useUpdateCurrentUser } from '../hooks/useUpdateCurrentUser';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    padding: theme.spacing.unit*2,
+    position:"relative"
+  },
+  assistance: { //UNUSED
+    color: theme.palette.text.normal,
+  },
+  'easy-going': {
+    color: theme.palette.text.moderationGuidelinesEasygoing,
+  },
+  'norm-enforcing': {
+    color: theme.palette.text.moderationGuidelinesNormEnforcing,
+  },
+  'reign-of-terror': {
+    color: theme.palette.text.moderationGuidelinesReignOfTerror,
+  },
+  'editButton': {
+    cursor: "pointer",
+    position: 'absolute',
+    right: 16,
+    height: '0.8em'
+  },
+  collapse: {
+    display:"flex",
+    justifyContent:"flex-end",
+    fontSize: 14,
+    marginBottom: 4,
+  },
+  moderationGuidelines: {
+    fontSize: "1.1rem",
+    '& p, & ul': {
+      marginTop: '.6em',
+      marginBottom: '.6em'
+    },
+    '& li': {
+      marginTop: '.4em',
+      marginBottom: '.4em'
+    }
+  }
+})
+
+const NewUserGuidelinesDialog = ({classes, onClose, post, user}: {
+  classes: ClassesType,
+  onClose: () => void,
+  post: PostsMinimumInfo,
+  user: UsersCurrent
+}) => {
+  const { LWDialog } = Components;
+  const updateCurrentUser = useUpdateCurrentUser();
+  const { recordEvent } = useNewEvents();
+
+  const handleClick = (e: React.MouseEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+
+    void updateCurrentUser({
+      acknowledgedNewUserGuidelines: true
+    });
+
+    const eventProperties = {
+      userId: user._id,
+      important: false,
+      intercom: true,
+      documentId: post._id,
+    };
+
+    recordEvent('acknowledged-new-user-guidelines', false, eventProperties);
```

Comment:
Huh, never seen this before

Reply:
I was originally just going to use the record in `LWEvent` via a resolver field to determine whether the user had ever acknowledged the new user guidelines, but it turns out to be quite annoying to then persist that fact on the client (to avoid the modal showing up multiple times in the same session).  Storing it as an non-resolver field on the user schema lets us use `updateCurrentUser`, which also updates the local cache.  But I think we might still plausibly want a record of when (and on what post) each user acknowledged the guidelines, without storing that information in a bloated nested object on the user.